py/visdom/server.py

```diff
@@ -840,6 +845,29 @@ def post(self):
         )
         self.wrap_func(self, args)
 
+class ForkHandler(BaseHandler):
+    def initialize(self, app):
+        self.app = app
+        self.state = app.state
+        self.login_enabled = app.login_enabled
+
+    @staticmethod
+    def wrap_func(handler, args):
+        prev_eid = args.get('prev_eid')
+        eid = args.get('eid')
+
+        if prev_eid not in handler.state:
+            return handler.write('TODO: env doesn\'t exit')
+
+        handler.state[eid] = copy.deepcopy(handler.state[prev_eid])
+        handler.write('TODO: missing proper reply')
```

Comment:
Proper reply here is probably either `True` or the new env_id. As the failure case will be an exception, there's no real requirement for the success case.